integrationExamples/gpt/top-level-paapi/tl_paapi_example.html

```diff
@@ -0,0 +1,186 @@
+                                            'auctionTimestamp': new Date().getTime(),
+                                            'publisherId': '537143056',
+                                            'adUnitId': '538703464'
+                                        },
+                                        'interestGroupBuyers': [
+                                            'https://privacysandbox.openx.net'
+                                        ],
+                                        'perBuyerSignals': {
+                                            'https://privacysandbox.openx.net': {
+                                                'bid': 1.5
+                                            }
+                                        },
+                                        'resolveToConfig': true,
+                                        'sellerCurrency': 'USD'
+                                    }
+                                ];
+                            }
+                        }
+                    ]
+                },
+            });
+
+            pbjs.addAdUnits(adUnits);
+            pbjs.requestBids({
+                bidsBackHandler: sendAdserverRequest,
+                timeout: PREBID_TIMEOUT
+            });
+        });
+
+        function raa() {
+            return Promise.all(
+                Object.entries(pbjs.getPAAPIConfig())
+                    .map(([adUnitCode, auctionConfig]) => {
+                        return navigator.runAdAuction({
+                            seller: window.location.origin,
+                            decisionLogicUrl: new URL('decisionLogic.js', window.location).toString(),
+                            resolveToConfig: false,
+                            ...auctionConfig
+                        }).then(urn => {
+                            if (urn) {
+                                // if we have a paapi winner, replace the adunit div
+                                // with an iframe that renders it
+                                const iframe = document.createElement('iframe');
+                                Object.assign(iframe, {
+                                    src: urn,
+                                    frameBorder: 0,
+                                    scrolling: 'no',
+                                }, auctionConfig.requestedSize);
+                                const div = document.getElementById(adUnitCode);
+                                div.parentElement.insertBefore(iframe, div);
+                                div.remove();
+                                return true;
+                            }
+                        });
+                    })
+            ).then(won => won.every(el => el));
+        }
+
+        function sendAdserverRequest() {
+            if (pbjs.adserverRequestSent) return;
+            pbjs.adserverRequestSent = true;
+            raa().then((allPaapi) => {
+                if (!allPaapi) {
```

Comment:
this calls gam if paapi doesn't fill, correct? But PAAPI will always fill if we give it the best contextual bid from prebid as its best alternative? Is the below code basically saying 'Let's compete Prebid vs PAAPI, and if neither provide any ad call gam?'

Reply:
I don't think we can give it the best contextual bid. Some bidders may decide to provide "extra" bids for it, but those don't necessarily contain all, or even the same, contextual bids, and Prebid can't control them.

Reply:
That's `auctionSignals.prebid.bidfloor` no ? I don't know if all buyers take it into account, so it may still not fill even if everyone bid all the time and above the floor.

Anyway, I assume that deleting the `<div>` is what makes GAM skip that slot if it was filled by PAAPI. Is there a better way to disable the slot more explicitly and perhaps less permanent ? It's fine either way as a test page but I imagine it won't work if I wanted to periodically refresh the ads.

Reply:
Yes, this is not a great way to render. I am not sure yet what the render API will look like:

 - when using GAM, the use case I'm hearing is to run fledge only after GAM picks the Prebid contextual winner (to keep adx demand in GAM). In practice this would mean returning the URN back to the rendering logic running in the GAM creative (PUC or equivalent). The publisher would not call any rendering API but somehow enable this behavior.
 - when not using GAM, the publisher calls `renderAd(document, adId)`. We could do something similar and only "pretend" to render `adId`, rendering the fledge ad instead in a nested frame, if the publisher enabled the same behavior as above. Or we could provide separate `runPaapi`/`renderPaapi` APIs. 
 - I'm not sure there will be a use case for what's demoed here, only run gam if fledge has no winner. Pubs would probably prefer to let GAM run fledge and get better demand.